angular-apollo-tailwind/src/app/profile/profile-repos-view/profile-repos.store.ts

```diff
@@ -135,19 +139,37 @@ export class ProfileReposStore extends ComponentStore<ProfileReposState> {
     private routeConfigService: RouteConfigService<string, 'profile'>,
     private userReposGQL: UserReposGQL,
     private orgReposGQL: OrgReposGQL,
+    private activatedRoute: ActivatedRoute,
+    private location: Location,
   ) {
     super(INITIAL_PROFILE_REPOS_STATE);
   }
 
   private getProfileRepos(owner: string, state: ProfileFilterState) {
+    const endCursor = this.activatedRoute.snapshot.queryParams['after'];
+    const startCursor = this.activatedRoute.snapshot.queryParams['before'];
+    const last = startCursor ? DEFAULT_CURSOR : undefined;
+    let first = endCursor ? DEFAULT_CURSOR : undefined;
+
+    if (endCursor === undefined && startCursor === undefined) {
+      first = DEFAULT_CURSOR;
+    }
+
+    this.location.replaceState(
+      this.location
+        .path()
+        .replace(`after=${endCursor}`, '')
+        .replace(`before=${startCursor}`, ''),
+    );
```

Comment:
I'm removing the params after they're consumed because if you start going to different tabs, etc. it breaks a lot of different pages because other stores want to consume these params

I'm not sure why we moved this to query params in the first place outside of that's how the React version does it -- do you know a way the Router can be utilized for this without refreshing the window?

Reply:
Yeah, not sure any way around this -- using something like this as a simple test:

```
this.router.navigate([], {
  relativeTo: this.activatedRoute,
  queryParams: {},
});
```

doesn't allow any pagination to occur, I think because the store/component is reloading and refetching the default initial state instead of preserving pagination state.

Reply:
This is now resolved: in `pagination.component.ts` I remove the params as such OnInit -- Then on button clicks the `after` or `before` params were not being remove which was causing a bug. I also removed all `location.replaceState` from the stores: 

```ts
ngOnInit(): void {
  this.router.navigate([], {
    queryParamsHandling: 'merge',
    queryParams: {
      after: null,
      before: null,
    },
  });
}

async handlePreviousPageClick(): Promise<boolean> {
  return await this.router.navigate([], {
    queryParamsHandling: 'merge',
    queryParams: {
      after: null,
      before: this.pageInfo?.startCursor,
    },
  });
}

async handleNextPageClick(): Promise<boolean> {
  return await this.router.navigate([], {
    queryParamsHandling: 'merge',
    queryParams: {
      after: this.pageInfo?.endCursor,
      before: null,
    },
  });
}
```